engine/execution/computation/result/consumer.go

```diff
@@ -0,0 +1,31 @@
+package result
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// ExecutedCollection holds results of a collection execution
+type ExecutedCollection interface {
+	// BlockHeader returns the block header in which collection was included
+	BlockHeader() *flow.Header
+
+	// Collection returns the content of the collection
+	Collection() *flow.Collection
+
+	// RegisterUpdates returns all registers that were updated during collection execution
+	UpdatedRegisters() flow.RegisterEntries
+
+	// TouchedRegisters returns all registers that has been touched (read/written) during collection execution
```

Comment:
can we move away from tracking the touch registers, and instead move toward tracking reads from the underlying storage?  you only the read set to re-execute the transactions

Reply:
That's not true, as of now you need the full register touches to verify execution on the verification nodes, given we have to provide proofs for the registers that are going to be written as well.

Reply:
the union of read set + updated set gives you the touch set.  there's no need to make the touch set first class

Reply:
yeah we could keep the API on primitives, changed it here [066683c](https://github.com/onflow/flow-go/pull/4049/commits/066683c63c4de690855a8e6443d9a51c769e12dc)